Feature/remove ddd concepts

DominikGuzei opened the pull request:
This PR implements the suggestions by @rhyslbw to reduce space:accounts to a simple wrapper around Meteor.Accounts – just accepting commands and sending out events so that other BCs can communicate with it.

DominikGuzei committed:
allow all combinations of username/email settings

DominikGuzei committed:
adds first draft of registration process

DominikGuzei committed:
updates to latest event-sourcing package

DominikGuzei committed:
let meteor create users again ;-)

DominikGuzei committed:
route login attempts to account aggregate

DominikGuzei committed:
mark password as sha256 digest before creating meteor users

DominikGuzei committed:
fixes typo

DominikGuzei committed:
make registration id optional for account messages

DominikGuzei committed:
remove dependency on Meteor.users

DominikGuzei committed:
fix the tests for the account aggregate

DominikGuzei committed:
reduce boilerplate

DominikGuzei committed:
improve and unify wording for login service

DominikGuzei committed:
refactors accounts test file

DominikGuzei committed:
updates to new event-routing and auto-correlation api

DominikGuzei committed:
simplifies and improves signup process

DominikGuzei committed:
removes users router

DominikGuzei committed:
updates to latest event-sourcing PR

DominikGuzei committed:
reduces package to meteor accounts wrapper

DominikGuzei committed:
fixes mgp dependencies

rhyslbw commented on local-packages.json:
```diff
@@ -0,0 +1,14 @@
+{
```
add to .gitignore

rhyslbw commented on source/server/publishers/authentication-publisher.js:
```diff
@@ -0,0 +1,32 @@
+    this.accounts.onLoginFailure(_.bind(this.handleFailedLogin, this));
+  },
+
+  handleFailedLogin(login) {
```
Maybe just move this into the onDependenciesReady block. No point in complicating things, and the testing will be based on event being published rather than method being called.

DominikGuzei replied:
yep and then i can also use the new ES6 fat arrow :wink:

rhyslbw commented:
I'm happy with this :+1:

DominikGuzei committed:
removes local-packages.json from repo

DominikGuzei committed:
make signup command configurable

and some other bugfixes and improvements, yeah
i know – not the best commit message in the world
